rest-service/manager_rest/test/endpoints/test_deployment_groups.py

```diff
@@ -437,6 +437,86 @@ def test_group_labels_for_deployments(self):
             {'label3': 'value4'},
         ])
 
+    def test_delete_group_label(self):
+        """Deleting a label from the group, deletes it from its deps"""
+        self.client.deployments.update_labels('dep1', [{'label1': 'value1'}])
+        self.client.deployment_groups.put(
+            'group1',
+            labels=[{'label1': 'value1'}],
+        )
+        self.client.deployment_groups.put(
+            'group1',
+            deployment_ids=['dep1']
+        )
+        group = self.client.deployment_groups.put(
+            'group1',
+            labels=[]
+        )
+        assert group.labels == []
+        client_dep = self.client.deployments.get('dep1')
+        assert client_dep.labels == []
+
+    def test_add_group_label(self):
+        """Adding a label to a group with deps, adds it to the deps"""
+        self.client.deployments.update_labels('dep1', [{'label1': 'value1'}])
+        group = self.client.deployment_groups.put(
+            'group1',
+            deployment_ids=['dep1'],
+        )
+        self.client.deployment_groups.put(
+            'group1',
+            labels=[{'label2': 'value2'}],
+        )
+        dep_id = group.deployment_ids[0]
+        client_dep = self.client.deployments.get(dep_id)
+        self.sm.get(models.Deployment, dep_id)
+        self.assert_resource_labels(client_dep.labels, [
+            {'label1': 'value1'}, {'label2': 'value2'}
+        ])
+
+    def test_add_labels_already_exist(self):
+        self.client.deployments.update_labels('dep1', [{'label2': 'value2'}])
+        self.client.deployment_groups.put(
+            'group1',
+            deployment_ids=['dep1'],
+        )
+        self.client.deployment_groups.put(  # doesn't throw
+            'group1',
+            labels=[{'label2': 'value2'}],
+        )
```

Comment:
Is it worth asserting on the what labels the deployment has here- a cheap operation but should alert us if we do something like duplicate labels with the same value?

Reply:
there's an unique index on them, but why not!